schedule/storetemporaryschedules.go

```diff
@@ -116,7 +117,12 @@ func (store *Store) updateFixedShifts(ctx context.Context, tx *sql.Tx, scheduleI
 		return err
 	}
 
-	rawData, err = json.Marshal(data)
+	if len(rawData) > 0 {
+		// preserve unknown fields
+		rawData, err = jsonutil.Apply(rawData, data)
+	} else {
+		rawData, err = json.Marshal(data)
+	}
```

Comment:
Is this check needed since in `Merge()` if the dst is empty, we handle creating a new map with just the new data?

Reply:
Correct, we use the new data as-is if there was nothing before and only do the merge if we had JSON to begin with.

Reply:
I mean, would we be able to do something like

```suggestion
rawData, err = jsonutil.Apply(rawData, data)
```

since the result should be the same?

Reply:
Just added a unit test, it handles when `rawData` is nil properly, so I'll remove the `if` check to simplify.